semantic-python/src/Language/Python/Core.hs

```diff
@@ -89,14 +100,14 @@ instance Compile (Py.ExecStatement Span)
 deriving via CompileSum (Py.Expression Span) instance Compile (Py.Expression Span)
 
 instance Compile (Py.ExpressionStatement Span) where
-  compile Py.ExpressionStatement { Py.extraChildren = children } = do
+  compile it@Py.ExpressionStatement { Py.extraChildren = children } = do
     actions <- traverse compile children
-    pure $ do' (fmap (Nothing :<-) actions)
+    pure . locate it $ do' (fmap (Nothing :<-) actions)
 
 instance Compile (Py.ExpressionList Span) where
-  compile Py.ExpressionList { Py.extraChildren = exprs } = do
+  compile it@Py.ExpressionList { Py.extraChildren = exprs } = do
     actions <- traverse compile exprs
-    pure $ do' (fmap (Nothing :<-) actions)
+    pure . locate it $ do' (fmap (Nothing :<-) actions)
 
 
 instance Compile (Py.False Span) where compile _ = pure (bool False)
```

Comment:
NB: you’re not calling `locate` here. One potential consequence is that stepping a debugger through a function returning a bool would relate the wrong span.